### DISCUSSION TITLE
Internal server error while trying to get uploaded file name in FastAPI

### DISCUSSION THREAD

**Discussion by iamrahulramesh** [fastapi/fastapi#7057] [Questions] [answered]
What I am trying is to upload an image using html template and to create a copy of the image by generating random names and to display the new name in a webpage.

```
from fastapi import FastAPI,File, UploadFile, Request
from fastapi.responses import HTMLResponse
from fastapi.staticfiles import StaticFiles
from fastapi.templating import Jinja2Templates
import shutil
import os
import datetime
import string
import random


A = datetime.datetime.now()
N = 10
ran = ''.join(random.choices(string.ascii_uppercase + string.digits, k= N))

           


app = FastAPI()

app.mount("/static", StaticFiles(directory="static"), name="static")


templates = Jinja2Templates(directory="templates")


@app.get("/uploadfile", response_class=HTMLResponse)
async def read_item(request: Request ):
    return templates.TemplateResponse("imageupload.html", {"request": request})

@app.post("/uploadfile")
async def create_image(request:Request,image: UploadFile = File(...)):
    abc = os.path.splitext(image.filename)[-1]
    print(abc)
    newname =image + str(A) + ran + abc
    print(newname)
    image.filename = newname
    imagepath = "/static/media/" + image.filename
    print(imagepath)

    with open(os.path.join("./static/media/",image.filename),"wb+") as buffer:
        shutil.copyfileobj(image.filename, buffer)
    return templates.TemplateResponse("imagename.html",{"request": request,"image_path":imagepath})
```

**Comment by Mause** [comment]
What is the error in your server logs?

**Comment by ycd** [comment]
Please edit your code with proper formatting, so others can easily test and debug your code.

**Comment by iamrahulramesh** [comment]
> What is the error in your server logs?

![image](https://user-images.githubusercontent.com/66358439/99936037-2a18c280-2d88-11eb-992c-94e7855067a1.png)
![image](https://user-images.githubusercontent.com/66358439/99936109-5a606100-2d88-11eb-9cb6-cbb627af652d.png)
![image](https://user-images.githubusercontent.com/66358439/99936131-664c2300-2d88-11eb-9e24-515b72ace286.png)

**Comment by iamrahulramesh** [comment]
> Please edit your code with proper formatting, so others can easily test and debug your code.

I think I did now. I am trying to  learn fastapi , What I am trying is to upload an image using html template and to create a copy of the image by generating random names and to  display the new name in a webpage

**Comment by ycd** [comment]
> > What is the error in your server logs?
> 
> ![image](https://user-images.githubusercontent.com/66358439/99936037-2a18c280-2d88-11eb-992c-94e7855067a1.png)
> ![image](https://user-images.githubusercontent.com/66358439/99936109-5a606100-2d88-11eb-9cb6-cbb627af652d.png)
> ![image](https://user-images.githubusercontent.com/66358439/99936131-664c2300-2d88-11eb-9e24-515b72ace286.png)

Can you share the error that is raised when you get an internal server error?

**Comment by iamrahulramesh** [comment]
> > > What is the error in your server logs?
> > 
> > 
> > ![image](https://user-images.githubusercontent.com/66358439/99936037-2a18c280-2d88-11eb-992c-94e7855067a1.png)
> > ![image](https://user-images.githubusercontent.com/66358439/99936109-5a606100-2d88-11eb-9cb6-cbb627af652d.png)
> > ![image](https://user-images.githubusercontent.com/66358439/99936131-664c2300-2d88-11eb-9e24-515b72ace286.png)
> 
> Can you share the error that is raised when you get an internal server error?

![image](https://user-images.githubusercontent.com/66358439/99948322-512ebe80-2d9f-11eb-9e15-d1fab8519bde.png)
This is my error in  servor logs

**Comment by iamrahulramesh** [comment]
> What is the error in your server logs?

![image](https://user-images.githubusercontent.com/66358439/99949093-7bcd4700-2da0-11eb-9eb3-e964beefd3a9.png)
This is my error in server logs

**Comment by Mause** [comment, marked as answer]
Please look at the documentation for UploadFile: https://fastapi.tiangolo.com/tutorial/request-files#uploadfile

**Reply by YuriiMotov** [reply]
`newname =image.filename + str(A) + ran + abc` (should be `image.filename` instead of `image`)

**Comment by iamrahulramesh** [comment]
My issue is solved now

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:

**Comment by hyshot** [comment]
Hi, can you tell me how you solved the problem? I have a code similar to this and I need to have an image uploaded statically.

**Comment by webdagger** [comment]
Please look at the documentation for UploadFile: https://fastapi.tiangolo.com/tutorial/request-files#uploadfile
